src/theory/quantifiers/quant_conflict_find.cpp

```diff
@@ -610,27 +621,52 @@ bool QuantInfo::setMatch(size_t v, TNode n, bool isGroundRep, bool isGround)
   Trace("qcf-match-debug") << "-- bind : " << v << " -> " << n << ", checked "
                            << d_curr_var_deq[v].size() << " disequalities"
                            << std::endl;
-  if (isGround)
+  if (isGround && d_vars[v].getKind() == BOUND_VARIABLE)
   {
-    if (d_vars[v].getKind() == BOUND_VARIABLE)
+    // Set the inst match object if this corresponds to an original variable
+    if (v < d_q[0].getNumChildren())
     {
-      d_vars_set.insert(v);
-      Trace("qcf-match-debug")
-          << "---- now bound " << d_vars_set.size() << " / "
-          << d_q[0].getNumChildren() << " base variables." << std::endl;
+      // we overwrite, so we must reset/set here
+      if (!d_instMatch.get(v).isNull())
+      {
+        d_instMatch.reset(v);
+      }
+      if (!d_instMatch.set(v, n))
+      {
+        return false;
+      }
     }
-  }
+    d_vars_set.insert(v);
+    Trace("qcf-match-debug")
+        << "---- now bound " << d_vars_set.size() << " / "
+        << d_q[0].getNumChildren() << " base variables." << std::endl;
+  }
+  // Note that assigning to a variable that an original variable is equal to
+  // should trigger the match object. For example, if we have auxiliary
+  // variable k and original variable x where x <-> k currently, and we set
+  // k -> t, then we should notify the match object that x -> t. However,
+  // this is not done here, as it would require more complex bookkeeping.
```

Comment:
clarify if it's done somewhere else. Also clarify the implications of not doing it if it's not done at all.